tar/multitape/multitape_write.c

```diff
@@ -416,6 +416,9 @@ writetape_open(uint64_t machinenum, const char * cachedir,
 		goto err0;
 	memset(d, 0, sizeof(struct multitape_write_internal));
 
+	/* Initialize lockfd as "not locked". */
+	d->lockfd = -1;
```

Comment:
I did consider omitting this part and changing the "check if ok to close" to:

```
if (d->lockfd > 0)
    close(d->lockfd);
```

but then I figured that if stdin was closed somewhere else (due to a new bug), then `multitape_lock(cachedir)` could return 0, which would then be interpreted as "not locked".  It seemed safer to go with -1, which is definitely not a real descriptor.